app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt

```diff
@@ -0,0 +1,133 @@
+        }
+        return SwapWidgetModel.Content(
+            isStatic = true,
+            widgetTitle = swapWidgetToTitle(),
+            availableAmount = null,
+            balance = balance(token),
+            currencyName = tokenName(token),
+            amount = tokenAmount(token, tokenAmount),
+            fiatAmount = fiatAmount,
+            amountMaxDecimals = token.decimals,
+        )
+    }
+
+    fun mapTokenBLoading(
+        token: SwapTokenModel,
+    ): SwapWidgetModel.Content =
+        SwapWidgetModel.Content(
+            isStatic = true,
+            widgetTitle = swapWidgetToTitle(),
+            currencyName = tokenName(token),
+            amount = TextViewCellModel.Skeleton(bigSkeleton()),
+            balance = balance(token),
+            fiatAmount = null,
+            availableAmount = null,
+        )
+
+    private fun fiatAmount(token: SwapTokenModel, tokenAmount: BigDecimal?): TextViewCellModel.Raw? {
+        if (tokenAmount == null) return null
+        val ratio = when (token) {
+            is SwapTokenModel.JupiterToken -> token.token.priceInUsd
+            is SwapTokenModel.UserToken -> token.token.totalInUsd
+        } ?: return null
+
+        val usd = tokenAmount.multiply(ratio).formatFiat()
+        return TextViewCellModel.Raw(TextContainer(R.string.swap_main_fiat_value, usd))
+    }
+
+    private fun tokenAmount(token: SwapTokenModel, tokenAmount: BigDecimal?): TextViewCellModel.Raw {
+        val decimals = token.decimals
+        val amountText = tokenAmount?.formatToken(decimals) ?: "0"
+        return TextViewCellModel.Raw(TextContainer(amountText))
+    }
+
+    private fun tokenName(token: SwapTokenModel): TextViewCellModel.Raw =
+        TextViewCellModel.Raw(
+            TextContainer(
+                when (token) {
+                    is SwapTokenModel.JupiterToken -> token.token.tokenName
+                    is SwapTokenModel.UserToken -> token.token.tokenName
+                }
+            )
+        )
+
+    private fun balance(token: SwapTokenModel): TextViewCellModel.Raw? =
+        when (token) {
+            is SwapTokenModel.JupiterToken -> null
+            is SwapTokenModel.UserToken -> balanceText(token)
+        }
+
+    private fun balanceText(token: SwapTokenModel.UserToken): TextViewCellModel.Raw =
+        TextViewCellModel.Raw(
+            TextContainer(R.string.swap_main_balance_amount, tokenAmount(token))
+        )
+
+    private fun availableAmount(token: SwapTokenModel): TextViewCellModel.Raw? =
+        when (token) {
+            is SwapTokenModel.JupiterToken -> null
+            is SwapTokenModel.UserToken -> availableAmountText(token)
+        }
+
+    private fun availableAmountText(token: SwapTokenModel.UserToken): TextViewCellModel.Raw =
+        TextViewCellModel.Raw(
+            TextContainer(R.string.swap_main_available_amount, tokenAmount(token))
+        )
+
+    private fun tokenAmount(token: SwapTokenModel.UserToken): String =
```

Comment:
can be as method  of UserToken

Reply:
UserToken is domain layer model, they not need formate it for ui. don't want overload like org.p2p.core.token.Token